source/extensions/filters/http/cache/config.cc

```diff
@@ -18,11 +18,13 @@ Http::FilterFactoryCb CacheFilterFactory::createFilterFactoryFromProtoTyped(
         fmt::format("Didn't find a registered implementation for type: '{}'", type));
   }
 
+  // Capture the cache instance as a reference; the factory must own it.
+  auto cache = std::ref(http_cache_factory->getCache(config, context));
```

Comment:
The factory is effectively static. Should we be using SingletonManager for this if it's costly?

/wait-any

Reply:
I'm not familiar enough with SingletonManager to understand how it would help (I'm also not sure if you're proposing that the cache should use SingletonManager or the factory should).

I *think* you're suggesting the factory should own nothing because it's static, and should instead instantiate its cache using SingletonManager? Or a set-of-caches in SingletonManager? I *think* that could work, and amusingly would be horribly broken *without* this change, because the SingletonManager doesn't own anything, so filters going out of scope would delete the cache with the previous implementation, but the lambda-capture is persistent until a config change so the cache would be mostly-persistent with this *and* changing getCache to std::shared_ptr *and* SingletonManager usage.

I'd be happy to make that larger change if the code-owners are amenable (I know this filter is like half in Google and half here so I'm reluctant to make larger changes without those eyes).

Reply:
Also changing getCache to return shared_ptr would be a breaking change for any existing third-party cache implementations.

Reply:
Yes the factory should not really own anything as it's fully static. Anything else should be created per-filter or using SingletonManager. I would just add a TODO capturing this discussion and we can fix it later.